drivers/char/gsochar.c

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file   	gsochar.c
+ * @author 	Deepak Khatri
+ * @date   	6 June 2020
+ * @version 0.1
+ * @brief   Warmp up task for GSoC2020.
+ * @see 	https://github.com/lorforlinux/gsoc-simple-char
+ */
+
+#include <linux/init.h>           // Macros used to mark up functions e.g. __init __exit
+#include <linux/module.h>         // Core header for loading LKMs into the kernel
+#include <linux/device.h>         // Header to support the kernel Driver Model
+#include <linux/kernel.h>         // Contains types, macros, functions for the kernel
+#include <linux/fs.h>             // Header for the Linux file system support
+#include <linux/uaccess.h>        // Required for the copy to user function
+
+#define  DEVICE_NAME "gsochar"    ///< The device will appear at /dev/gsochar using this value
+#define  CLASS_NAME  "gsoc"       ///< The device class -- this is a character device driver
+#define BUFFER_SIZE 1024
+
+MODULE_LICENSE("GPL");            ///< The license type -- this affects available functionality
+MODULE_AUTHOR("Deepak khatri");    ///< The author -- visible when you use modinfo
+MODULE_DESCRIPTION("A simple Linux char driver for the GSoc warmup task.");  ///< The description -- see modinfo
+MODULE_VERSION("0.1");
+
+static int    majorNumber;                  ///< Stores the device number -- determined automatically
+static char   message[BUFFER_SIZE] = {0};   ///< Memory for the string that is passed from userspace
+static int    numberOpens = 0;              ///< Counts the number of times the device is opened
+static struct class*  gsocharClass  = NULL; ///< The device-driver class struct pointer
+static struct device* gsocharDevice = NULL; ///< The device-driver device struct pointer
+static int placeholder = 0;
+
+/** @brief The device open function that is called each time the device is opened
+ *  This will only increment the numberOpens counter in this case.
+ *  @param inodep A pointer to an inode object (defined in linux/fs.h)
+ *  @param filep A pointer to a file object (defined in linux/fs.h)
+ */
+static int dev_open(struct inode *inodep, struct file *filep){
+   numberOpens++;
+   printk(KERN_INFO "gsochar: Device has been opened %d time(s)\n", numberOpens);
```

Comment:
I am very new to kernel programming and I thought instead of using pr_info() it would be a good idea to use the longer version just to learn about different kernel log message levels.